pkg/volume/flexvolume/unmounter.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flexvolume
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/util/exec"
+	"k8s.io/kubernetes/pkg/volume"
+	"k8s.io/kubernetes/pkg/volume/util"
+)
+
+// FlexVolumeUnmounter is the disk that will be cleaned by this plugin.
+type flexVolumeUnmounter struct {
+	*flexVolume
+	// Runner used to teardown the volume.
+	runner exec.Interface
+	volume.MetricsNil
+}
+
+var _ volume.Unmounter = &flexVolumeUnmounter{}
+
+// Unmounter interface
+func (f *flexVolumeUnmounter) TearDown() error {
+	path := f.GetPath()
+	return f.TearDownAt(path)
+}
+
+func (f *flexVolumeUnmounter) TearDownAt(dir string) error {
+
+	if pathExists, pathErr := util.PathExists(dir); pathErr != nil {
```

Comment:
Ditto the `UnmountDevice` command comment:  I suggest removing this logic. If a driver implements this call they can implement these checks. If they don't then the default `TearDownAt` command should just call UnmountPath(...) from pkg/volume/util/util.go which has the most robust implementation of this logic.